### DISCUSSION TITLE
Optimal way to check if client is connected before emitting, due to bad namespace error?

### DISCUSSION THREAD

**Discussion by Mathiaszero** [miguelgrinberg/python-socketio#1402] [Q&A]
What's the best way to check that the client is connected for the following situation? I get `socketio.exceptions.BadNamespaceError: / is not a connected namespace.` which is probably due to emitting when the client is not connected to the server.

**Comment by miguelgrinberg** [comment]
If a namespace is connected then `client.get_sid(namespace)` will return a value of type string. If the namespace is not connected, the return will be `None`. You can also catch the `BadNamespaceError`. Which way is best really depends on each particular situation and your personal preferences.
